base_layer/core/src/covenants/decoder.rs

```diff
@@ -127,13 +127,41 @@ mod test {
     use super::*;
     use crate::{
         covenant,
-        covenants::{arguments::CovenantArg, fields::OutputField, filters::CovenantFilter},
+        covenants::{
+            arguments::CovenantArg,
+            byte_codes::ARG_OUTPUT_FIELD,
+            fields::OutputField,
+            filters::CovenantFilter,
+        },
     };
 
     #[test]
     fn it_immediately_ends_iterator_given_empty_bytes() {
         let buf = &[] as &[u8; 0];
         assert!(CovenantTokenDecoder::new(&mut &buf[..]).next().is_none());
+        assert!(CovenantTokenDecoder::new(&mut &buf[..]).next().is_none());
```

Comment:
Why do this twice?

Reply:
Checks the iterator behaviour, that it continues to return None - not necessary, removed